components/raftstore-v2/src/fsm/peer.rs

```diff
@@ -95,18 +113,103 @@ impl<EK: KvEngine, ER: RaftEngine> Fsm for PeerFsm<EK, ER> {
 }
 
 pub struct PeerFsmDelegate<'a, EK: KvEngine, ER: RaftEngine, T> {
-    fsm: &'a mut PeerFsm<EK, ER>,
-    store_ctx: &'a mut StoreContext<T>,
+    pub fsm: &'a mut PeerFsm<EK, ER>,
+    pub store_ctx: &'a mut StoreContext<EK, ER, T>,
 }
 
-impl<'a, EK: KvEngine, ER: RaftEngine, T> PeerFsmDelegate<'a, EK, ER, T> {
-    pub fn new(fsm: &'a mut PeerFsm<EK, ER>, store_ctx: &'a mut StoreContext<T>) -> Self {
+impl<'a, EK: KvEngine, ER: RaftEngine, T: Transport> PeerFsmDelegate<'a, EK, ER, T> {
+    pub fn new(fsm: &'a mut PeerFsm<EK, ER>, store_ctx: &'a mut StoreContext<EK, ER, T>) -> Self {
         Self { fsm, store_ctx }
     }
 
-    pub fn handle_msgs(&self, peer_msgs_buf: &mut Vec<PeerMsg<EK>>) {
+    pub fn schedule_tick(&mut self, tick: PeerTick) {
```

Comment:
The names are not intuitive enough IMO (or need more documentations), to handle a tick we go through four functions:
```
- handle_tick(tick)
	- on_raft_tick
		- register_raft_tick
			- schedule_tick(tick)
```

But from the names of these steps, I can't tell the difference or their relationships.

Reply:
handle_tick - handle any type of ticks
on_raft_tick - only handle raft tick
register_raft_tick - only register raft tick
schedule_tick - schedule any type of tick

There are many types of tick, raft tick is just one of them.

Reply:
I know, that's why I added `(tick)` in my pseudo code. The problem is the "verb", in your explanation, it seems they should all be called handle except for `schedule_tick`.